src/types.ts

```diff
@@ -1,7 +1,19 @@
+import type { WarpCoreConfig } from '@hyperlane-xyz/sdk';
 import { z } from 'zod';
 
 // https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-5.html#the-awaited-type-and-promise-improvements
 export type MaybePromise<T> = T | Promise<T> | PromiseLike<T>;
 
 export const ChainAddressesSchema = z.record(z.string());
 export type ChainAddresses = z.infer<typeof ChainAddressesSchema>;
+
+export type WarpRouteId = string;
+export type WarpRouteConfigMap = Record<WarpRouteId, WarpCoreConfig>;
+
+export type RecursivePartial<T> = {
```

Comment:
woah
```ts
type DeepPartial<T> = T extends object ? {
    [P in keyof T]?: DeepPartial<T[P]>;
} : T;
```
does this work too? from https://stackoverflow.com/questions/61132262/typescript-deep-partial

Reply:
Not sure that would handle arrays in objects - like `tokens` in a warp config - but I'll try. If it works that's definitely nicer!

Reply:
also not sure
I feel like Array might extend object?

Reply:
Works!